File: 5/Grains/src/HelloWorld.cs
using System.Threading.Tasks;
using Interfaces;
using Library;

namespace Grains
{
    public class HelloWorld : Orleans.Grain, IHelloWorld
    {
        public Task<string> SayHello(string name)
        {
            return Task.FromResult(Say.hello(name));
        }
    }
}

File: 5a/Grains/src/HelloWorld.fs
namespace Grains

open System.Threading.Tasks
open Interfaces
open Library
open Orleans

type HelloWorld() =
    inherit Grain()

    interface IHelloWorld with
        member this.SayHello (name: string) (_: GrainCancellationToken) : Task<string> =
            task { return Say.hello (name) }
